docs/topics/gradle.md

```diff
@@ -927,12 +927,17 @@ Here is a complete list of options for Gradle tasks:
 
 ### Attributes common to JVM, JS, and JS DCE
 
-| Name | Description | Possible values |Default value |
-|------|-------------|-----------------|--------------|
-| `allWarningsAsErrors` | Report an error if there are any warnings |  | false |
-| `suppressWarnings` | Don't generate warnings |  | false |
-| `verbose` | Enable verbose logging output. Works only when the [Gradle debug log level enabled](https://docs.gradle.org/current/userguide/logging.html) |  | false |
-| `freeCompilerArgs` | A list of additional compiler arguments |  | [] |
+| Name | Description                                                                                                                                         | Possible values |Default value |
+|------|-----------------------------------------------------------------------------------------------------------------------------------------------------|-----------------|--------------|
+| `allWarningsAsErrors` | Report an error if there are any warnings                                                                                                           |  | false |
+| `suppressWarnings` | Don't generate warnings                                                                                                                             |  | false |
+| `verbose` | Enable verbose logging output. Works only when the [Gradle debug log level enabled](https://docs.gradle.org/current/userguide/logging.html)         |  | false |
+| `freeCompilerArgs` | A list of additional compiler arguments. You can use experimental `-X` arguments here too. See an [example](#example-of-additional-arguments-usage) |  | [] |
+
+> We are  going to remove the attribute `freeCompilerArgs` in future releases. If you miss some option in the Kotlin Gradle DSL, 
```

Comment:
> remove

This option will stay in some other form, but will produce build warning on any usage. Better to replace it with "to discourage `freeCompilerArgs` usage in future releases"?

Reply:
What about "We are going to deprecate..."? We should tell the reason why we discourage something. And if we say that something will deprecate – it's obvious that it's better not to use it.

Reply:
`freeCompilerArgs` has following issues:
- values that could be passed are not discoverable and user has to find them by searching in the internet
- values does not have proper deprecation cycle notifying user in advance that it is deprecated
- passing options with paths could break build cache via using absolute paths. Generally some values could use Gradle inputs specific normalization
- some passed values should not be considered as task inputs as they are not affect compilation output
- not nice build scan report for task inputs comparison between builds